### ISSUE TITLE
Helm `version` param doesn't exist in released CRD

### ISSUE THREAD

**Issue by rltvty** [argoproj/argo-cd#4468]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a
question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

The documentation states that you can specify the Helm version to use in the declarative setup for an Application.  However, the `version` flag is stripped by Kubernetes on deploy.

In the latest release, the CRD does not have the `version` parameter: https://github.com/argoproj/argo-cd/blob/v1.7.7/manifests/crds/application-crd.yaml#L198-L208

But in the master version of the CRD, it does exist: https://github.com/argoproj/argo-cd/blob/master/manifests/crds/application-crd.yaml#L198-L211

Due to this, if I set the parameter `version` on the application (with no other helm parameters set), Kubernetes strips the field from the manifest.  But Argo sees that it exists in source control, so it re-applies the manifest.  And then Kubernetes strips it again.  And repeat... in a constant sync loop.

See related PR where the `version` parameter was added: https://github.com/argoproj/argo-cd/pull/4111

**Expected behavior**

The released CRD should allow a `version` parameter for `helm`, to match docs.  

**Version**

```shell
argocd: v1.7.7+33c93ae
  BuildDate: 2020-09-29T04:54:55Z
  GitCommit: 33c93aea0b9ee3d02fb9703cd82cecce3540e954
  GitTreeState: clean
  GoVersion: go1.14.1
  Compiler: gc
  Platform: linux/amd64
```

**Comment by jessesuen** [comment]
This because the feature only exists in master and is not yet released. v1.7 does not have this feature.

**Comment by mars64** [comment]
> This because the feature only exists in master and is not yet released. v1.7 does not have this feature.

How is a user supposed to know this?

* the [example application spec in the documentation](https://github.com/argoproj/argo-cd/blame/91e62bfc3b98b12cd7d0baba6c082edbf82fdc34/docs/operator-manual/application.yaml#L56) indicates this feature is available, makes no mention of version requirements, and is [actively referred to by argocd docs
](https://argoproj.github.io/argo-cd/operator-manual/declarative-setup/)
* [the userguide site](https://argoproj.github.io/argo-cd/user_guide/) is crazy broken (recommended reading as per https://argoproj.github.io/argo-cd/#getting-started)

**Comment by rltvty** [comment]
Things that aren't released should not be in docs!

**Comment by jannfis** [comment]
> How is a user supposed to know this?

I agree that we have some serious debt regarding documentation. We were already thinking of migrating the docs to a system that actually can handle versioned docs (such as readthedocs.org), so that users will be able to use the documentation for the specific version of Argo CD they are using. Currently, we are pretty much cluttering the docs with version information (available since vX, or only before vY etc).

Personally, I think the debt goes further than that, and that the docs could also need some serious love and restructuring. However, this is obviously a lot of work, and everyone is invited to contribute.

>  * [the userguide site](https://argoproj.github.io/argo-cd/user_guide/) is crazy broken (recommended reading as per https://argoproj.github.io/argo-cd/#getting-started)

Oops, that is due to the fact that indeed the link is pointing to the wrong location. Correct link would be https://argoproj.github.io/argo-cd/user-guide/, I've submitted a fix with #4477. Thanks for bringing that to our attention.

**Comment by mars64** [comment]
> I agree that we have some serious debt regarding documentation. We were already thinking of migrating the docs to a system that actually can handle versioned docs (such as readthedocs.org), so that users will be able to use the documentation for the specific version of Argo CD they are using. Currently, we are pretty much cluttering the docs with version information (available since vX, or only before vY etc).
> 
> Personally, I think the debt goes further than that, and that the docs could also need some serious love and restructuring. However, this is obviously a lot of work, and everyone is invited to contribute.

Huge thank you for the acknowledgement here, I think I'm not alone in my expectations of the docs site and knowing that the group at least has it on the radar is of much value to me. Perhaps I'll take a look at that [contribution guide](https://argoproj.github.io/argo-cd/developer-guide/contributing/)! (subtle plug eh ;) )

> > * [the userguide site](https://argoproj.github.io/argo-cd/user_guide/) is crazy broken (recommended reading as per https://argoproj.github.io/argo-cd/#getting-started)
> 
> Oops, that is due to the fact that indeed the link is pointing to the wrong location. Correct link would be https://argoproj.github.io/argo-cd/user-guide/, I've submitted a fix with #4477. Thanks for bringing that to our attention.

Thank you!

**Comment by Stephan3555** [comment]
Is there any timeline when and in which version this feature will be released?

**Comment by mcwienczek** [comment]
I think it was released with version 1.8.0: 
https://github.com/argoproj/argo-cd/releases/tag/v1.8.0
